programs/ssl/ssl_client2.c

```diff
@@ -1949,7 +2006,22 @@ int main( int argc, char *argv[] )
         }
 #endif
 
-    mbedtls_ssl_conf_rng( &conf, mbedtls_ctr_drbg_random, &ctr_drbg );
+    if (opt.reproducible)
+    {
+        srand( 1 );
+        mbedtls_ssl_conf_rng( &conf, dummy_random, &ctr_drbg );
+#if defined(MBEDTLS_PLATFORM_TIME_ALT)
+        mbedtls_platform_set_time( dummy_constant_time );
```

Comment:
If `MBEDTLS_HAVE_TIME` is not set, we don't need to `mbedtls_platform_set_time( dummy_constant_time );`, do we?

If `MBEDTLS_HAVE_TIME` is set, we also need `MBEDTLS_PLATFORM_TIME_ALT` to be set, so we can swap in the `dummy_constant_time`. Otherwise, we don't need to.

I think I'd find this a bit easier to follow:

```C
#if defined(MBEDTLS_HAVE_TIME)
#if defined(MBEDTLS_PLATFORM_TIME_ALT)
        mbedtls_platform_set_time( dummy_constant_time );
#else
        fprintf( stderr, "Warning: reproducible option used without constant time\n" );
#endif
#endif
```

Would you agree?

Reply:
> If MBEDTLS_HAVE_TIME is not set, we don't need to mbedtls_platform_set_time( dummy_constant_time );, do we?

I think we do not.

> Would you agree?

I agreed and committed, thanks for reviewing